util/types.ts

```diff
@@ -14,7 +11,9 @@ export type EnsureLookup<
   },
 > = Lookup;
 
-export type Flatten<T> = T extends Function ? T : { [K in keyof T]: T[K] };
+export type Flatten<T> = T extends (infer E)[] ? Flatten<E>[]
+  : T extends object ? { [K in keyof T]: Flatten<T[K]> }
+  : T;
```

Comment:
This breaks `Flatten` for tuples. I've recently adopted
```ts
export type Flatten<T> = [{ [K in keyof T]: Flatten<T[K]> }][0]
```
with the caveat that it should only be used to wrap things known to be simple objects or arrays